File: data/derived_cot/rq1_traces/translation/glm5.1/py_to_cpp/code_output/BankAccount.cpp
#include <stdexcept>

class BankAccount {
private:
    int balance;

public:
    BankAccount(int balance = 0) : balance(balance) {}

    int deposit(int amount) {
        if (amount < 0) {
            throw std::invalid_argument("Invalid amount");
        }
        balance += amount;
        return balance;
    }

    int withdraw(int amount) {
        if (amount < 0) {
            throw std::invalid_argument("Invalid amount");
        }
        if (amount > balance) {
            throw std::invalid_argument("Insufficient balance.");
        }
        balance -= amount;
        return balance;
    }

    int view_balance() const {
        return balance;
    }

    void transfer(BankAccount& other_account, int amount) {
        withdraw(amount);
        other_account.deposit(amount);
    }
};

File: data/derived_cot/rq1_traces/translation/glm5.1/java_to_py/code_output/BankAccount.py
class BankAccount:
    def __init__(self, balance=0):
        # Using _balance to indicate it's intended to be private, matching the Java code's intent
        self._balance = balance

    def deposit(self, amount):
        if amount < 0:
            raise ValueError("Invalid amount")
        self._balance += amount
        return self._balance

    def withdraw(self, amount):
        if amount < 0:
            raise ValueError("Invalid amount")
        if amount > self._balance:
            raise ValueError("Insufficient balance.")
        self._balance -= amount
        return self._balance

    def view_balance(self):
        return self._balance

    def transfer(self, other_account, amount):
        self.withdraw(amount)
        other_account.deposit(amount)

if __name__ == "__main__":
    account1 = BankAccount()
    account2 = BankAccount()
    account1.deposit(1000)
    account1.transfer(account2, 300)
    print(f"account1.balance = {account1.view_balance()}")
    print(f"account2.balance = {account2.view_balance()}")